pkg/pillar/zedcloud/tls.go

```diff
@@ -172,6 +172,30 @@ func GetTlsConfig(dns *types.DeviceNetworkStatus, serverName string, clientCert
 	return tlsConfig, nil
 }
 
+// UpdateTLSProxyCerts - Update when DeviceNetworkStatus changes
+func UpdateTLSProxyCerts(ctx *ZedCloudContext) {
+	tlsCfg := ctx.TlsConfig
+	devNS := ctx.DeviceNetworkStatus
+	if tlsCfg == nil || devNS == nil {
+		log.Errorln("UpdateTLSProxyCerts: tlsconfig or dev NS missing")
+		return
+	}
+
+	var updated bool
+	caCertPool := tlsCfg.RootCAs
+	// AppendCertsFromPEM checks duplicates inside
+	for _, port := range devNS.Ports {
+		for _, pem := range port.ProxyCertPEM {
+			caCertPool.AppendCertsFromPEM(pem)
+			updated = true
+		}
+	}
```

Comment:
Don't you need code to delete any proxy certs which are now no longer part of the config?

Reply:
Well, I here assume changing proxy certs very rarely happen, and keep the old ones does not hurt until the next reboot. Otherwise, we can replace this function with the normal GetTlsConfig() if we worry about this.

Reply:
If folks change a proxy root cert eg due to concerns that the private key might have been compromised then it would be quite bad if the devices continue to trust it. Thus on a change or delete of a proxy cert I think you need to rebuild the list from scratch. Adds work well with the code you have

Reply:
If folks change a proxy root cert eg due to concerns that the private key might have been compromised then it would be quite bad if the devices continue to trust it. Thus on a change or delete of a proxy cert I think you need to rebuild the list from scratch. Adds work well with the code you have

Reply:
Ok. I can build a cache in the zedcloudContext for the proxyCerts, and compare with the DNS update new one and decide to add, skip or rebuid